src/EventSubscriber/ConfigSubscriber.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\field_encrypt\EventSubscriber\ConfigSubscriber.
+ */
+
+namespace Drupal\field_encrypt\EventSubscriber;
+
+use Drupal\Core\Config\ConfigCrudEvent;
+use Drupal\Core\Config\ConfigEvents;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\Query\QueryFactory;
+use Drupal\Core\Queue\QueueFactory;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\Core\StringTranslation\TranslationInterface;
+use Drupal\Core\Url;
+use Drupal\field\Entity\FieldStorageConfig;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Updates existing data when field encryption settings are updated.
+ */
+class ConfigSubscriber implements EventSubscriberInterface {
+  use StringTranslationTrait;
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * The entity query service.
+   *
+   * @var \Drupal\Core\Entity\Query\QueryFactory
+   */
+  protected $entityQuery;
+
+  /**
+   * The queue factory.
+   *
+   * @var \Drupal\Core\Queue\QueueFactory
+   */
+  protected $queueFactory;
+
+
+  /**
+   * Constructs a new ConfigSubscriber object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_manager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Entity\Query\QueryFactory $entity_query
+   *   The entity query service.
+   * @param \Drupal\Core\Queue\QueueFactory $queue_factory
+   *   The queue factory.
+   * @param \Drupal\Core\StringTranslation\TranslationInterface $translation
+   *   The string translation service.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_manager, QueryFactory $entity_query, QueueFactory $queue_factory, TranslationInterface $translation) {
+    $this->entityManager = $entity_manager;
+    $this->entityQuery = $entity_query;
+    $this->queueFactory = $queue_factory;
+    $this->stringTranslation = $translation;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    $events[ConfigEvents::SAVE][] = array('onConfigSave', 0);
```

Comment:
Ok so there are a few delete scenarios:

* A ConfigEvents::DELETE event gets triggered - however this will only be triggered if the whole field (config) gets deleted. Thus there will be no more field data left, since this means the field gets deleted.
At this point there might be leftover EncryptedFieldValue entities - so I should add logic to delete those. The existing field storage will already be deleted by core.

* A hook_entity_delete() gets triggered - this already deletes the linked EncryptedFieldValue entities in the current codebase. The existing field storage will already be deleted by core.

* A field was encrypted, but storage settings are changed to no longer be encrypted. This will trigger a ConfigEvents::SAVE event. The current codebase will make sure that the existing field storage gets filled with the plain text data again, and EncryptedFieldValue entities will be deleted (in \Drupal\field_encrypt\FieldEncryptProcessEntities::updateStoredField()